caas/kubernetes/provider/application/application.go

```diff
@@ -1089,6 +1090,40 @@ func (a *app) Service() (*caas.Service, error) {
 	}, nil
 }
 
+func (a *app) computeStatus(ctx context.Context, client kubernetes.Interface, now time.Time) (string, status.Status, time.Time, error) {
+	jujuStatus := status.Waiting
+	switch a.deploymentType {
+	case caas.DeploymentStateful:
+		ss, err := a.getStatefulSet()
+		if err != nil {
+			return "", jujuStatus, now, errors.Trace(err)
+		}
+		if ss.GetDeletionTimestamp() != nil {
+			jujuStatus = status.Terminated
+		} else if ss.Spec.Replicas != nil && ss.Status.ReadyReplicas == *ss.Spec.Replicas {
+			jujuStatus = status.Active
+		}
+		var statusMessage string
+		if jujuStatus == status.Waiting {
+			events, err := ss.Events(ctx, client)
+			if err != nil {
+				return "", jujuStatus, now, errors.Trace(err)
+			}
+			// Take the most recent event.
+			if count := len(events); count > 0 {
+				evt := events[count-1]
+				if evt.Type == corev1.EventTypeWarning && evt.Reason == "FailedCreate" {
+					jujuStatus = status.Blocked
```

Comment:
error is what we'd want here.
blocked is usually reserved as being for the charm to set

Reply:
This logic was copied from operator k8s code. 
yeah, agreed, it should be Error.